lib/arjdbc/jdbc/adapter.rb

```diff
@@ -289,12 +289,14 @@ def active?
 
       # @override
       def reconnect!
+        super
```

Comment:
Yeah, they must have cleaned it up or something. Here is the relevant code from the abstract adapter:

```
      # Disconnects from the database if already connected, and establishes a
      # new connection with the database. Implementors should call super if they
      # override the default implementation.
      def reconnect!
        clear_cache!
        reset_transaction
      end

      # Disconnects from the database if already connected. Otherwise, this
      # method does nothing.
      def disconnect!
        clear_cache!
        reset_transaction
      end

      ###
      # Clear any caching the database adapter may be doing, for example
      # clearing the prepared statement cache. This is database specific.
      def clear_cache!
        # this should be overridden by concrete adapters
      end

      def reset_transaction #:nodoc:
        @transaction_manager = ConnectionAdapters::TransactionManager.new(self)
      end
```

The issue was that the transaction state wasn't being updated, so the adapter would think it was still inside an active transaction even though the connection had been closed.